src/lib/app.ts

```diff
@@ -190,4 +193,33 @@ export class AppObject {
                 });
         });
     }
+
+    public async sufficientRightsBookmark(rights: string[]): Promise<boolean> {
+
+        const title = "SerOnDemandCheckRightBookmark";
+        let checker = true;
+
+        let bookmarkProperties: EngineAPI.IGenericBookmarkProperties = {
+            qInfo: {
+                qType: "hiddenbookmark"
+            },
+            qMetaDef: {
+                title: title,
+                approved: false
+            },
+            creationDate: (new Date()).toISOString()
+        };
+
+        const bookmarkObject = await this.app.createBookmark(bookmarkProperties);
+        const bookmarkLayout = await bookmarkObject.getLayout();
+
+        rights.forEach((right) => {
```

Comment:
you can shorten this one and search only for first value of that

```ts
/** should be the same but breaks after first match and returns true (converted to false) */
checker = !rights.some((right) => (bookmarkLayout.qMeta as IMetaBookmarkExtended).privileges.indexOf(right) === -1);
```

Reply:
fixed with https://github.com/senseexcel/ser-ext-ondemand/commit/20834c08ae836aeaa95d2c8de8465dacbb87c2df